pkg/updater/resultstore/resultstore_test.go

```diff
@@ -93,6 +95,61 @@ func timeMustText(t time.Time) string {
 	return string(s)
 }
 
+func TestExtractGroupID(t *testing.T) {
+	cases := []struct {
+		name string
+		tg   *configpb.TestGroup
+		pr   *processedResult
+		want string
+	}{
+		{
+			name: "nil",
+		},
+		{
+			name: "primary grouping BUILD by override config value",
+			tg: &configpb.TestGroup{
+				DaysOfResults:                   7,
+				BuildOverrideConfigurationValue: "test-key-1",
+				PrimaryGrouping:                 configpb.TestGroup_PRIMARY_GROUPING_BUILD,
+			},
+			pr: &processedResult{
+				InvocationProto: &resultstore.Invocation{
+					Id: &resultstore.Invocation_Id{
+						InvocationId: "id-1",
+					},
+					Properties: []*resultstore.Property{
+						{
+							Key:   "test-key-1",
+							Value: "test-val-1",
+						},
+					},
+					Name: invocationName("id-1"),
+					Timing: &resultstore.Timing{
+						StartTime: &timestamppb.Timestamp{
+							Seconds: 1234,
+						},
+					},
+				},
+			},
+			want: "test-val-1",
+		},
```

Comment:
please add a test case for any fallback grouping which will default to invocation ID